component/prometheus/exporter/exporter.go

```diff
@@ -134,6 +108,44 @@ func (c *Component) Update(args component.Arguments) error {
 	return err
 }
 
+// Handler serves metrics endpoint from the integration implementation.
+func (c *Component) Handler() http.Handler {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	return c.metricsHandler
+}
+
+func newExporter(creator Creator, name string, targetBuilderFunc func(discovery.Target, component.Arguments) []discovery.Target) func(component.Options, component.Arguments) (component.Component, error) {
+	return func(opts component.Options, args component.Arguments) (component.Component, error) {
+		c := &Component{
+			opts:              opts,
+			reload:            make(chan struct{}, 1),
+			creator:           creator,
+			targetBuilderFunc: targetBuilderFunc,
+		}
+		jobName := fmt.Sprintf("integrations/%s", name)
+		instance := defaultInstance()
+
+		c.baseTarget = discovery.Target{
+			model.AddressLabel:                  opts.HTTPListenAddr,
+			model.SchemeLabel:                   "http",
+			model.MetricsPathLabel:              path.Join(opts.HTTPPath, "metrics"),
+			"instance":                          instance,
+			"job":                               jobName,
+			"__meta_agent_integration_name":     jobName,
+			"__meta_agent_integration_instance": instance,
+			"__meta_agent_integration_id":       opts.ID,
```

Comment:
note: this is a new label

Reply:
would `__meta_agent_component_id` be better?

Reply:
I will add `__meta_agent_hostname` (we used to have `agent_hostname` in static mode)